packages/start/src/server-handler/index.tsx

```diff
@@ -12,40 +12,108 @@ import {
+  const regex = new RegExp(
+    `${sanitizeBase(process.env.TSS_SERVER_FN_BASE)}/([^/?#]+)`,
+  )
+
+  // Execute the regex
+  const match = url.pathname.match(regex)
+  const serverFnId = match ? match[1] : null
   const search = Object.fromEntries(url.searchParams.entries()) as {
-    _serverFnId?: string
-    _serverFnName?: string
     payload?: any
   }
 
-  const serverFnId = search._serverFnId
-  const serverFnName = search._serverFnName
-
-  if (!serverFnId || !serverFnName) {
-    throw new Error('Invalid request')
+  if (typeof serverFnId !== 'string') {
+    throw new Error('Invalid server action param for serverFnId: ' + serverFnId)
   }
 
-  invariant(typeof serverFnId === 'string', 'Invalid server action')
+  const serverFnInfo = serverFnManifest[serverFnId]
+
+  if (!serverFnInfo) {
+    console.log('serverFnManifest', serverFnManifest)
+    throw new Error('Server function info not found for ' + serverFnId)
+  }
 
   if (process.env.NODE_ENV === 'development')
-    console.info(`ServerFn Request: ${serverFnId} - ${serverFnName}`)
-  if (process.env.NODE_ENV === 'development') console.info()
+    console.info(`\nServerFn Request: ${serverFnId}`)
+
+  let fnModule: undefined | { [key: string]: any }
+
+  if (process.env.NODE_ENV === 'development') {
+    fnModule = await (globalThis as any).app
+      .getRouter('server')
+      .internals.devServer.ssrLoadModule(serverFnInfo.extractedFilename)
+  } else {
+    fnModule = await serverFnInfo.importer()
+  }
 
-  const action = (await getManifest('server').chunks[serverFnId]?.import())?.[
-    serverFnName
-  ] as Function
+  // let moduleUrl = serverFnInfo.extractedFilename
+  // // In dev, we (for now) use Vinxi to get the "server" server-side router
+  // // Then we use that router's devServer.ssrLoadModule to get the serverFn
+  // if (process.env.NODE_ENV === 'development') {
+  //   fnModule = await (globalThis as any).app
+  //     .getRouter('server')
+  //     .internals.devServer.ssrLoadModule(serverFnInfo.extractedFilename)
+  // } else {
+  //   // In prod, we use the serverFn's chunkName to get the serverFn
+  //   const router = (globalThis as any).app.getRouter('server')
+  //   const filePath = join(
+  //     router.outDir,
+  //     router.base,
+  //     serverFnInfo.chunkName + '.mjs',
+  //   )
+  //   moduleUrl = pathToFileURL(filePath).toString()
+  //   fnModule = await import(/* @vite-ignore */ moduleUrl)
+  // }
+
+  if (!fnModule) {
+    console.log('serverFnManifest', serverFnManifest)
+    throw new Error('Server function module not resolved for ' + serverFnId)
+  }
+
+  const action = fnModule[serverFnInfo.functionName]
+
+  if (!action) {
+    console.log('serverFnManifest', serverFnManifest)
+    console.log('fnModule', fnModule)
+    throw new Error(
+      `Server function module export not resolved for serverFn ID: ${serverFnId}`,
+    )
+  }
 
   const response = await (async () => {
     try {
```

Comment:
This may be outside the scope of this PR, but since I noticed it, I figured I'd bring it up.

Currently, here we directly importing `defaultTransformer` from `@tanstack/react-router` and consuming. What happens when the user has declared the custom transformer like SuperJSON ([as mentioned here in the docs](https://tanstack.com/router/latest/docs/framework/react/guide/ssr#data-transformers))?

Perhaps this can be solved since we require the user to pass in a `createRouter` function in the `app/routes/ssr.tsx` file which can be used here in the server 🤔... Question is, are there any changes that need to happen on the client fetcher...